packages/ui-components/src/components/ListVisual/ListVisual.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { FlexSpaceBetween } from '../Layout';
+import { ColorCircle } from './ColorCircle';
+import { HeaderRow, SubHeaderRow, StandardRow, DrillDownRow } from './Rows';
+import { FetchLoader } from '../FetchLoader';
+
+const Container = styled.div`
+  border-radius: 3px;
+  overflow: hidden;
+  min-height: 400px;
+  height: ${props => (props.isLoading ? '400px' : 'auto')};
+  background: #f9f9f9;
+`;
+
+const Footer = styled(FlexSpaceBetween)`
+  padding: 2rem 1rem;
+`;
+
+const ROW_TYPE_COMPONENTS = {
+  header: HeaderRow,
+  subheader: SubHeaderRow,
+  standard: StandardRow,
+};
+
+const VALUE_TYPE_COMPONENTS = {
+  color: ColorCircle,
+};
+
+const DEFAULT_LIST_CONFIGS = {
+  COLOR: {
+    0: { color: '#ffc701', label: 'Yellow' },
+    1: { color: '#02B851', label: 'Green' },
+    '-1': { color: '#D13333', label: 'Red' },
+  },
+};
+
+const getDisplayConfig = ({ valueType, listConfig }, statistic) => {
+  switch (valueType) {
+    case 'color':
+      if (listConfig) {
+        return listConfig[statistic];
+      }
+      return DEFAULT_LIST_CONFIGS.COLOR[statistic] || null;
+    default:
+      return null;
+  }
+};
+
+const processData = (config, data, drillDowns) => {
+  let parentCode = null;
+
+  return data?.map(item => {
+    let rowType = 'standard';
+    let displayConfig = null;
+    let drillDownCode = null;
+
+    if (!item.parent) {
+      rowType = 'header';
+      parentCode = item.code;
+    }
+
+    if (item.parent === parentCode) {
+      rowType = 'subheader';
+    }
+
+    if (config?.drillDown?.itemCodeByEntry[item.code] !== undefined) {
+      drillDownCode = config?.drillDown?.itemCodeByEntry[item.code];
+
+      // check that the drill down config exists before trying to render a drilldown row
+      if (drillDowns.find(d => d.code === drillDownCode)) {
+        rowType = 'drilldown';
+      }
+    }
+
+    if (item.statistic !== undefined) {
+      displayConfig = getDisplayConfig(config, item.statistic);
+    }
+
+    return { ...item, rowType, valueType: config.valueType, displayConfig, drillDownCode };
+  });
+};
+
+export const ListVisual = React.memo(
+  ({ viewContent, isLoading, dashboard, isError, error, drillDowns, DrillDownComponent }) => {
```

Comment:
The `DrillDownComponent` is passed in as a prop so that the ListVisual can be de-coupled from the rest of the UI and moved into storybook. In future we may want to use this component in web-frontend or other apps and this is setup to handle that. It does make the component slightly more complex though...